File: app/src/main/java/com/mcal/disassembler/nativeapi/DisassemblerDumper.java
package com.mcal.disassembler.nativeapi;

public class DisassemblerDumper
{
	public static native void load(String path);
	public static native boolean hasFile(String path);
	public static native String getNameAt(long pos);
	public static native String getDemangledNameAt(long pos);
	public static native int getTypeAt(long pos);
	public static native int getBindAt(long pos);
	public static native long getSize();
	public static native String demangle(String name);
	public static native String demangleOnly(String name);
}

File: app/src/main/java/com/mcal/disassembler/nativeapi/DisassemblerDumper.kt
package com.mcal.disassembler.nativeapi

object DisassemblerDumper {
    external fun load(path: String)
    external fun hasFile(path: String): Boolean
    external fun getNameAt(pos: Long): String
    external fun getDemangledNameAt(pos: Long): String?
    external fun getTypeAt(pos: Long): Int
    external fun getBindAt(pos: Long): Int
    external fun getSize(): Long
    external fun demangle(name: String): String
    external fun demangleOnly(name: String): String
}